app/helpers/application_helper/toolbar/mixins/custom_button_toolbar_mixin.rb

```diff
@@ -0,0 +1,60 @@
+module ApplicationHelper::Toolbar::Mixins::CustomButtonToolbarMixin
+  APPLIES_TO_CLASS_BASE_MODELS = %w(AvailabilityZone CloudNetwork CloudObjectStoreContainer CloudSubnet CloudTenant
+                                    CloudVolume ContainerGroup ContainerImage ContainerNode ContainerProject
+                                    ContainerTemplate ContainerVolume EmsCluster ExtManagementSystem
+                                    GenericObject GenericObjectDefinition Host LoadBalancer
+                                    MiqGroup MiqTemplate NetworkRouter OrchestrationStack
+                                    PersistentVolume SecurityGroup Service
+                                    ServiceTemplate Storage Switch Tenant User Vm VmOrTemplate).freeze
+
+  def custom_button_appliable_class?(model)
+    model = "MiqTemplate" if model == "Image"
+    model = "Vm" if model == "Instance"
+    model = "ExtManagementSystem" if model == "StorageManager"
```

Comment:
this may need some code for the other managers .. thinking openstack providers probably also link to their network managers, etc.